forge/lang/expander.rkt

```diff
@@ -764,14 +767,45 @@
    (syntax/loc stx
     (begin
       (pred pwd.prop-name pwd.prop-expr)
-      (begin pwd.where-blocks ...)
       (test
         test_name
         #:preds [imp_total]
         #:scope pwd.scope
         #:bounds pwd.bounds
         #:expect theorem )))]))
 
+
+;; Quick and dirty static check to ensure a test
+;; references a predicate.
+
+;; Flattens test AST to a string and ensures that the
+;; referenced predicate's name is present.
+;; Could be done better by traversing the AST.
+;; This would help break apart test-expect
+;; blocks and examine each test.
+
+(define-for-syntax (ensure-target-ref target-pred)
+
+  (define tp (format "~a" (syntax->datum target-pred)))
+  (lambda (ex) 
+    (let ([ex-as-datum (syntax->datum ex)])
+      (unless 
+        (memq (syntax-e target-pred) (flatten ex-as-datum))
+        (eprintf  "Warning: ~a ~a:~a Test does not reference ~a." 
+          (syntax-source ex) (syntax-line ex) (syntax-column ex)  tp)))))
+
+
+(define-syntax (TestSuiteDecl stx)
+  (syntax-parse stx
+  [tsd:TestSuiteDeclClass 
+   
+    ;; Static checks on test blocks go here.
+   #:do [(for ([tc (syntax->list #'(tsd.test-constructs ...))])
```

Comment:
The `#:do` is optional; the for loop can go right above the `syntax/loc` templace.